### PR TITLE
Add python prebuilt wheels workflow

### PR DESCRIPTION
**Pull request by bingyanglin**
# Description of change

Add python wheels building flow.

## Links to any relevant issues

N/A

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Check the generated files by Github Action are expected.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/python_binding_publish.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    steps:
+     - uses: actions/checkout@v2
+     - uses: actions-rs/toolchain@v1
+       with:
+          toolchain: stable
+          override: true
+     - uses: actions/setup-python@v2
+       with:
+         python-version: ${{ matrix.python-version }}
+     - name: Build wheels
```
Comment (rajivshah3) [inline .github/workflows/python_binding_publish.yml:33]:
```suggestion
     - name: Install required packages
        run: |
          sudo apt-get update
          sudo apt-get install libudev-dev libusb-1.0-0-dev
          
     - name: Build wheels
```

File: .github/workflows/python_binding_publish.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+       run: |
+         choco install openssl --no-progress
+         echo "OPENSSL_DIR=C:\Program Files\OpenSSL-Win64" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+
+     - name: Build wheels
+       working-directory: bindings/python/native
+       run: |
+         python -m pip install maturin
+         maturin build --release --manylinux off
+     - uses: actions/upload-artifact@v1
```
Comment (rajivshah3) [inline .github/workflows/python_binding_publish.yml]: Is `actions/upload-artifact@v2` not working properly?

File: .github/workflows/python_binding_publish.yml (hunk 1 above)
Comment (bingyanglin) [reply .github/workflows/python_binding_publish.yml:33]: Thanks. Modified.

File: .github/workflows/python_binding_publish.yml (hunk 2 above)
Comment (bingyanglin) [reply .github/workflows/python_binding_publish.yml]: Should be. Let me try.

**Review by rajivshah3 — APPROVED** [review]

**Comment by rajivshah3** [comment]
Just remembered one last thing - should we statically link OpenSSL and ship it with the Python wheels instead of dynamically linking?

**Comment by bingyanglin** [comment]
> Just remembered one last thing - should we statically link OpenSSL and ship it with the Python wheels instead of dynamically linking?

I think good to have static linking. It seems that we can achieve this by using `sudo apt-get install glibc-static`?

**Comment by rajivshah3** [comment]
> It seems that we can achieve this by using sudo apt-get install `glibc-static`?

We just need to do a couple of hacks to get the `paho-mqtt` crate to use static linking for OpenSSL:

https://github.com/iotaledger/wallet.rs/blob/develop/.github/workflows/covector-version-or-publish.yml#L109-L127

We also need to use a fork because there are issues upstream:

```toml
[patch.'crates-io']
paho-mqtt = { git = "https://github.com/rajivshah3/paho.mqtt.rust", branch = "feature/openssl-static-link-master-0.9"}
```

File: .github/workflows/python_binding_publish.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      - master
+      - develop
+
+jobs:
+
+  linux-wheels:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (rajivshah3) [inline .github/workflows/python_binding_publish.yml:15]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9]
    env:
      OPENSSL_LINK_STATIC: true
```

File: .github/workflows/python_binding_publish.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+     - uses: actions/upload-artifact@v2
+       with:
+         name: linux-iota-client-py${{ matrix.python-version }}-wheel
+         path: bindings/python/native/target/wheels/
+
+  osx-wheels:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (rajivshah3) [inline .github/workflows/python_binding_publish.yml:47]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9]
    env:
      OPENSSL_LINK_STATIC: true
```

File: .github/workflows/python_binding_publish.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+     - uses: actions/upload-artifact@v2
+       with:
+         name: osx-iota-client-py${{ matrix.python-version }}-wheel
+         path: bindings/python/native/target/wheels/
+
+  windows-wheels:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (rajivshah3) [inline .github/workflows/python_binding_publish.yml:75]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9]
    env:
      OPENSSL_LINK_STATIC: true
```

**Review by rajivshah3 — COMMENTED** [review]
Sorry, forgot one more thing

File: .github/workflows/python_binding_publish.yml (hunk 3 above)
Comment (bingyanglin) [reply .github/workflows/python_binding_publish.yml:15]: Thanks. Modifying.

File: .github/workflows/python_binding_publish.yml (hunk 5 above)
Comment (bingyanglin) [reply .github/workflows/python_binding_publish.yml:75]: Thanks. Modifying.

File: .github/workflows/python_binding_publish.yml (hunk 4 above)
Comment (bingyanglin) [reply .github/workflows/python_binding_publish.yml:47]: Thanks. Modifying.

**Comment by rajivshah3** [comment]
Crap, I think the `Build and run tests` workflow needs to be adjusted because of the `paho` fork. I think if we add this like we did for the Node.js bindings it should succeed:

https://github.com/iotaledger/wallet.rs/blob/6f990839e194926c6b586937f54353994f713039/.github/workflows/build-and-test.yml#L163-L181

Sorry about this 😞

**Comment by bingyanglin** [comment]
> Crap, I think the `Build and run tests` workflow needs to be adjusted because of the `paho` fork. I think if we add this like we did for the Node.js bindings it should succeed:
> 
> https://github.com/iotaledger/wallet.rs/blob/6f990839e194926c6b586937f54353994f713039/.github/workflows/build-and-test.yml#L163-L181
> 
> Sorry about this

Nono don't be. Thank you for pointing this out. Let me modify this.

**Review by rajivshah3 — APPROVED** [review]
Looks good!

**Pull request merged**